packages/app/obojobo-repository/shared/components/repository-nav.test.js

```diff
@@ -169,4 +203,128 @@ describe('RepositoryNav', () => {
+		document.cookie = null
+
+		const reusableComponent = <RepositoryNav {...navProps} />
+		let component
+		act(() => {
+			component = create(reusableComponent)
+		})
+		const tree = component.toJSON()
+		expect(tree).toMatchSnapshot()
+
+		if (document && document.cookie) {
+			//don't get here
+		} else {
+			expect(document.cookie).toBe(null)
+		}
+		document.cookie = originalDocument
+	})
+
+	test('toggles notifications popup on button click', () => {
+		document.cookie =
+			'notifications=' +
+			JSON.stringify([
+				{ key: 1, text: 'Notification1', title: 'Title1' },
+				{ key: 2, text: 'Notification2', title: 'Title2' }
+			])
+
+		const component = create(<RepositoryNav {...navProps} />)
+		act(() => {
+			component.update(<RepositoryNav {...navProps} />)
+		})
+
+		const mockClickEvent = { preventDefault: jest.fn() }
+
+		act(() => {
+			component.root
+				.findByProps({ className: 'repository--nav--current-user--name' })
+				.children[1].props.onClick(mockClickEvent)
+			component.update(<RepositoryNav {...navProps} />)
+		})
+		expectNotificationsPopupToBeOpen(component)
+
+		act(() => {
+			component.root.findByProps({ className: 'exit-button' }).props.onClick()
+			component.update(<RepositoryNav {...navProps} />)
+		})
+		expectNotificationsPopupToBeClosed(component)
+	})
+
+	test('handles notification data from Notification component', () => {
+		document.cookie =
+			'notifications=' +
+			JSON.stringify([
+				{ key: 1, text: 'Notification1', title: 'Title1' },
+				{ key: 2, text: 'Notification2', title: 'Title2' }
+			])
+
+		const component = create(<RepositoryNav {...navProps} />)
+		act(() => {
+			component.update(<RepositoryNav {...navProps} />)
+		})
+
+		const mockClickEvent = { preventDefault: jest.fn() }
+
+		act(() => {
+			component.root
+				.findByProps({ className: 'repository--nav--current-user--name' })
+				.children[1].props.onClick(mockClickEvent)
+			component.update(<RepositoryNav {...navProps} />)
+		})
+
+		const notificationComponentInstance = component.root.findByType(Notification)
+
+		// Manually call the onDataFromNotification prop with some test data, this normally happens notificationIsOpen is true and Notifications are rendered
+		act(() => {
+			notificationComponentInstance.props.onDataFromNotification(5)
+		})
+
+		expect(
+			component.root.findByProps({ className: 'repository--nav--current-user--name' }).children[1]
+				.props.children[0]
+		).toBe(5)
+	})
+	test('renders null when there are no notifications but document.cookie is not null', () => {
+		const reusableComponent = <RepositoryNav {...navProps} />
+		let component
+		act(() => {
+			component = create(reusableComponent)
+		})
+		const tree = component.toJSON()
+		expect(tree).toMatchSnapshot()
+
+		if (document && document.cookie) {
+			const cookiePropsRaw = decodeURIComponent(document.cookie).split(';')
+
+			cookiePropsRaw.forEach(c => {
+				const parts = c.trim().split('=')
+
+				if (parts[0] === 'notifications') {
```

Comment:
See previous comment re: using `expect` to fail early instead of relying on an if/else to check for success.